kamelets/kafka-source.kamelet.yaml

```diff
@@ -134,4 +143,12 @@ spec:
         autoOffsetReset: "{{autoOffsetReset}}"
         groupId: "{{?consumerGroup}}"
       steps:
-      - to: "kamelet:sink"
+        - set-property:
+            name: deserializeHeaders
+            constant: "{{deserializeHeaders}}"
+        - choice:
+            when:
+              - simple: "${exchangeProperty.deserializeHeaders} == 'true'"
+                steps:
+                  - bean: "org.apache.camel.kamelets.utils.serialization.kafka.KafkaHeaderDeserializer"
```

Comment:
this may be added directly as a processor using local bean and it can be enabled/disabled without using choice

Reply:
@lburgazzoli have updated the Kamelet and processor. Is that what you have had in mind?

Reply:
But this is a minor improvement, I'm fine with what did

Reply:
wow that's cool! very smooth solution. let me use this

Reply:
@lburgazzoli I have tried to use this solution but getting errors:
```
org.apache.camel.NoSuchBeanException: No bean could be found in the registry for: #bean:{{kafkaHeaderDeserializer}} of type: org.apache.camel.Processor
[1] 	at org.apache.camel.support.CamelContextHelper.mandatoryLookup(CamelContextHelper.java:241)
[1] 	at org.apache.camel.reifier.AbstractReifier.mandatoryLookup(AbstractReifier.java:145)
[1] 	at org.apache.camel.reifier.ProcessReifier.createProcessor(ProcessReifier.java:40)
[1] 	at org.apache.camel.reifier.ProcessorReifier.makeProcessor(ProcessorReifier.java:838)
[1] 	at org.apache.camel.reifier.ProcessorReifier.addRoutes(ProcessorReifier.java:579)
[1] 	at org.apache.camel.reifier.RouteReifier.doCreateRoute(RouteReifier.java:236)
[1] 	at org.apache.camel.reifier.RouteReifier.createRoute(RouteReifier.java:74)
[1] 	at org.apache.camel.impl.DefaultModelReifierFactory.createRoute(DefaultModelReifierFactory.java:49)
[1] 	at org.apache.camel.impl.DefaultCamelContext.startRouteDefinitions(DefaultCamelContext.java:851)
[1] 	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.createRouteForEndpoint(KameletComponent.java:405)
[1] 	at org.apache.camel.component.kamelet.KameletComponent$LifecycleHandler.onContextInitialized(KameletComponent.java:419)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.doInit(AbstractCamelContext.java:2781)
[1] 	at org.apache.camel.quarkus.core.FastCamelContext.doInit(FastCamelContext.java:165)
[1] 	at org.apache.camel.support.service.BaseService.init(BaseService.java:83)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.init(AbstractCamelContext.java:2490)
[1] 	at org.apache.camel.support.service.BaseService.start(BaseService.java:111)
[1] 	at org.apache.camel.impl.engine.AbstractCamelContext.start(AbstractCamelContext.java:2509)
[1] 	at org.apache.camel.impl.DefaultCamelContext.start(DefaultCamelContext.java:246)
[1] 	at org.apache.camel.quarkus.main.CamelMain.doStart(CamelMain.java:94)
[1] 	at org.apache.camel.support.service.BaseService.start(BaseService.java:119)
[1] 	at org.apache.camel.quarkus.main.CamelMain.startEngine(CamelMain.java:139)
[1] 	at org.apache.camel.quarkus.main.CamelMainRuntime.start(CamelMainRuntime.java:49)
[1] 	at org.apache.camel.quarkus.core.CamelBootstrapRecorder.start(CamelBootstrapRecorder.java:45)
[1] 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy_0(CamelBootstrapProcessor$boot-173480958.zig:100)
[1] 	at io.quarkus.deployment.steps.CamelBootstrapProcessor$boot-173480958.deploy(CamelBootstrapProcessor$boot-173480958.zig:40)
[1] 	at io.quarkus.runner.ApplicationImpl.doStart(ApplicationImpl.zig:842)
[1] 	at io.quarkus.runtime.Application.start(Application.java:101)
[1] 	at io.quarkus.runtime.ApplicationLifecycleManager.run(ApplicationLifecycleManager.java:105)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:67)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:41)
[1] 	at io.quarkus.runtime.Quarkus.run(Quarkus.java:120)
[1] 	at io.quarkus.runner.GeneratedMain.main(GeneratedMain.zig:29)
[1] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[1] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[1] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[1] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[1] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.doRun(QuarkusEntryPoint.java:53)
[1] 	at io.quarkus.bootstrap.runner.QuarkusEntryPoint.main(QuarkusEntryPoint.java:28)
```
I can not see what I am doing wrong here

Reply:
use process + ref as in my example, that should work.

Your bean need to implement Processor

Reply:
ah found it - not allowed to use `#bean:` prefix in process:ref

Reply:
that did the trick, thx

Reply:
the fine smell of success 😄